proxycore/clientconn.go

```diff
@@ -85,7 +85,7 @@ func ConnectClient(ctx context.Context, endpoint Endpoint, config ClientConnConf
 
 func (c *ClientConn) Handshake(ctx context.Context, version primitive.ProtocolVersion, auth Authenticator) (primitive.ProtocolVersion, error) {
 	for {
-		response, err := c.SendAndReceive(ctx, frame.NewFrame(version, -1, message.NewStartup()))
+		response, err := c.SendAndReceive(ctx, frame.NewFrame(version, -1, message.NewStartup(message.StartupOptionCompression, "LZ4")))
```

Comment:
This tagged along with another change. Will need to figure out something else since we can't blindly set compression for every client between proxy and server.